api/src/main/java/jakarta/enterprise/concurrent/ZonedTrigger.java

```diff
@@ -0,0 +1,163 @@
+     * <P>
+     * This method is provided to maintain compatibility with {@link Trigger} and should not be
+     * implemented. The default implementation delegates to the method signature that
+     * accepts and returns <code>ZonedDateTime</code>.
+     *
+     * @param lastExecutionInfo information about the last execution of the task. 
+     *                          This value will be null if the task has not yet run.
+     * @param taskScheduledTime the date/time at which the
+     *                          {@code ManagedScheduledExecutorService.schedule}
+     *                          method was invoked to schedule the task.
+     * @return the date/time after which the next execution of the task should start.
+     * @throws IllegalArgumentException if the next run time is too large to represent as a <code>Date</code>.
+     */
+    default Date getNextRunTime(LastExecution lastExecutionInfo, Date taskScheduledTime) {
+        ZonedDateTime nextTime = getNextRunTime(lastExecutionInfo, taskScheduledTime.toInstant().atZone(getZoneId()));
+        return Date.from(nextTime.toInstant());
+    }
+
+    /**
+     * Returns the timezone to use for the
+     * {@link java.time.ZonedDateTime ZonedDateTime} that is supplied to the
+     * {@link #getNextRunTime(LastExecution, java.time.ZonedDateTime) getNextRunTime} and
+     * {@link #skipRun(LastExecution, java.time.ZonedDateTime) skipRun} methods.
+     * <P>
+     * The default implementation returns the system default timezone
+     * and should be overridden whenever there is chance that the server
+     * might not be running with the same timezone for which the
+     * business logic within this trigger is written.
+     *
+     * @return timezone to use for operations on this trigger.
+     */
+    default ZoneId getZoneId() {
+        return ZoneId.systemDefault();
+    }
+
+    /**
+     * Return true if this run instance should be skipped.
+     * <p>
+     * This is useful if the task shouldn't run because it is late or if the task 
+     * is paused or suspended.
+     * <p>
+     * Once this task is skipped, the state of its Future's result will throw a 
+     * {@link SkippedException}. Unchecked exceptions will be wrapped in a 
+     * <code>SkippedException</code>.
+     * <p>
+     * The default implementation returns <code>false</code>, making it optional to
+     * implement this method if you do not require support for skipped executions.
+     *
+     * @param lastExecutionInfo information about the last execution of the task. 
+     *                          This value will be null if the task has not yet run.
+     * @param scheduledRunTime  the date/time after which the execution of the task
+     *                          is scheduled to start.
+     * @return true if the task should be skipped and rescheduled.
+     */
+    default boolean skipRun(LastExecution lastExecutionInfo, ZonedDateTime scheduledRunTime) {
```

Comment:
does this need to be public?

Reply:
The interface is public, so these methods are implicitly public.  I'd be happy to explicitly add `public` to it for clarity.